### PR TITLE
CONSOLE-4019: Display a warning message from kube-apiserver when creating resource integration tests

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
### Screencast:

https://github.com/openshift/console/assets/15249132/819db043-f644-47a0-ab5f-6b94e99ca3e6

### Electron Screencast:

https://github.com/openshift/console/assets/15249132/9053d594-cd98-401f-984d-bc1ccaa710c3

### CODE REVIEW

File: frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        req.continue((res) => {
+          res.headers = {
+            Warning: WARNING_FOO,
+          };
+        });
+      }).as(POD_CREATED_ALIAS);
+      yamlEditor.clickSaveCreateButton();
+      cy.wait(`@${POD_CREATED_ALIAS}`, WAIT_OPTION);
+      detailsPage.sectionHeaderShouldExist('Pod details');
+      cy.byTestID(LEARN_MORE_ID).parents().contains('Admission Webhook Warning');
```
Comment (rhamilto) [inline frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts:134]: Use [data-test](https://github.com/openshift/console/blob/a4e560ac5d50c94fc36b351d109bc1db63968121/frontend/packages/console-shared/src/components/toast/ToastProvider.tsx#L61) on the `<Alert>` itself.

File: frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      cy.byTestID(LEARN_MORE_ID)
```
Comment (rhamilto) [inline frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts:135]: Use [data-test](https://github.com/openshift/console/blob/a4e560ac5d50c94fc36b351d109bc1db63968121/frontend/packages/console-shared/src/components/toast/ToastProvider.tsx#L61) on the `<Alert>` itself.

File: frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                Warning: resource.warning,
+              };
+            });
+          },
+        ).as(BULK_RESOURCES_CREATED_ALIAS);
+      }
+      yamlEditor.clickSaveCreateButton();
+      cy.wait(`@${BULK_RESOURCES_CREATED_ALIAS}`, WAIT_OPTION);
+      cy.byTestID('resources-successfully-created').contains('Resources successfully created');
+      cy.byTestID(LEARN_MORE_ID).parents().contains('Admission Webhook Warning');
```
Comment (rhamilto) [inline frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts:162]: Use [data-test](https://github.com/openshift/console/blob/a4e560ac5d50c94fc36b351d109bc1db63968121/frontend/packages/console-shared/src/components/toast/ToastProvider.tsx#L61) on the `<Alert>` itself.

File: frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      cy.byTestID(LEARN_MORE_ID)
```
Comment (rhamilto) [inline frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts:163]: Use [data-test](https://github.com/openshift/console/blob/a4e560ac5d50c94fc36b351d109bc1db63968121/frontend/packages/console-shared/src/components/toast/ToastProvider.tsx#L61) on the `<Alert>` itself.

File: frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        .parents()
+        .contains(`Pod ${POD_NAME}-b violates policy ${WARNING_FOO}`);
+      cy.byTestID(LEARN_MORE_ID)
```
Comment (rhamilto) [inline frontend/packages/integration-tests-cypress/tests/app/admission-webhook-warning-notifications.cy.ts:166]: Use [data-test](https://github.com/openshift/console/blob/a4e560ac5d50c94fc36b351d109bc1db63968121/frontend/packages/console-shared/src/components/toast/ToastProvider.tsx#L61) on the `<Alert>` itself.

**Review by rhamilto — COMMENTED** [review]
Looks good, but recommend using the `data-test` on the `<Alert>` itself in order to eliminate the dependency on `.parents()`.

**Comment by rhamilto** [comment]
One suggestion after watching your video (thanks for including that!):  run your tests locally using Electron since that is what is in use in CI.

**Comment by cyril-ui-developer** [comment]
> data-test

@rhamilto I thought of that earlier but the [Toast Context](https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/toast/ToastContext.tsx#L6) doesn't have `data-test` props.  Maybe I should just extend the props by adding `data-test`, right?

**Comment by rhamilto** [comment]
> > data-test
> 
> @rhamilto I thought of that earlier but the [Toast Context](https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/toast/ToastContext.tsx#L6) doesn't have `data-test` props. Maybe I should just extend the props by adding `data-test`, right?

Agreed.

File: frontend/packages/console-shared/src/components/toast/ToastProvider.tsx [hunk 6]
```diff
@@ -58,7 +58,7 @@ const ToastProvider: React.FC = ({ children }) => {
               variant={toast.variant}
               timeout={toast.timeout}
               onTimeout={() => removeToast(toast.id)}
-              data-test={`${toast.title} alert`}
+              data-test={toast.dataTest || 'alert'}
```
Comment (cyril-ui-developer) [inline frontend/packages/console-shared/src/components/toast/ToastProvider.tsx:61]: I need to run the full e2e tests to check if this change impact or break any existing test(s).

Comment (rhamilto) [reply frontend/packages/console-shared/src/components/toast/ToastProvider.tsx:61]:
```suggestion
              data-test={toast.dataTest || `${toast.title} alert`}
```

**Review by rhamilto — APPROVED** [review]
Nice work, @cyril-ui-developer 

/lgtm

**Comment by cyril-ui-developer** [comment]
Docs Approver:
/assign [@opayne1](https://github.com/opayne1)
PX Approver:
/assign [@RickJWagner](https://github.com/RickJWagner)
QA Approver:
/assign @yapei

**Comment by cyril-ui-developer** [comment]
Docs Approver:
/assign [@opayne1](https://github.com/opayne1)
PX Approver:
/assign [@RickJWagner](https://github.com/RickJWagner)

**Comment by cyril-ui-developer** [comment]
> Docs Approver: /assign [@opayne1](https://github.com/opayne1) PX Approver: /assign [@RickJWagner](https://github.com/RickJWagner)
/assign @opayne1 
/assign @RickJWagner

**Pull request merged**